assessment/sources/common.go

```diff
@@ -0,0 +1,25 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package common
+
+import "github.com/GoogleCloudPlatform/spanner-migration-tool/assessment/utils"
+
+type InfoSchema interface {
+	GetIndexInfo(dbName, table string) ([]utils.IndexAssessment, error)
```

Comment:
I suggest the interface be made more generic to cover things that are available in conv as well.